File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    var answer = '';
    participant.sort();
    completion.sort();
    const findName=()=>{
        for(let i=0;i<participant.length;i++){
            if(participant[i]!==completion[i]) {
                 return participant[i];
            }
        }
       return participant[participant.length-1]
    }
    answer=findName();
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
#include <map>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    map<string,int> m;
    // sort(participant.begin(),participant.end());
    // sort(completion.begin(),completion.end());
    // answer = participant[participant.size()-1];
    for(int i =0;i<completion.size();i++){
        m[completion[i]]++;
    }
    for(int i =0;i<participant.size();i++){
        if(m[participant[i]] == 0){ 
            return participant[i];
        }
        else m[participant[i]]--;
    }
    return answer;
}